internal/component/database_observability/postgres/component.go

```diff
@@ -366,20 +366,47 @@ func (c *Component) startCollectors(systemID string, engineVersion string) error
 
 	entryHandler := addLokiLabels(loki.NewEntryHandler(c.handler.Chan(), func() {}), c.instanceKey, systemID)
 
+	var tableRegistry *collector.TableRegistry
 	collectors := enableOrDisableCollectors(c.args)
 
+	if collectors[collector.SchemaDetailsCollector] {
+		stCollector, err := collector.NewSchemaDetails(collector.SchemaDetailsArguments{
+			DB:              c.dbConnection,
+			DSN:             string(c.args.DataSourceName),
+			CollectInterval: c.args.SchemaDetailsArguments.CollectInterval,
+			CacheEnabled:    c.args.SchemaDetailsArguments.CacheEnabled,
+			CacheSize:       c.args.SchemaDetailsArguments.CacheSize,
+			CacheTTL:        c.args.SchemaDetailsArguments.CacheTTL,
+			EntryHandler:    entryHandler,
+			Logger:          c.opts.Logger,
+		})
+		if err != nil {
+			logStartError(collector.SchemaDetailsCollector, "create", err)
+		}
+		if stCollector != nil {
+			tableRegistry = stCollector.GetTableRegistry()
```

Comment:
tableRegistry needs to be reassigned here by the Schema Details collector in the case that it is enabled so that the Query Details collector receives an updated reference.

Tried to add a test to protect this order, but could not successfully inject a mocked DB (have run into this problem before). Happy to take suggestions or look into this as a follow-up.